TLS 1.3: Documentation update for 3.6 release (#8930)
Mbed-TLS/mbedtls: docs/architecture/tls13-support.md

Overview
--------

> Mbed TLS provides an implementation of the TLS 1.3. The TLS 1.3 support

Reviewer: Wording does not quite work. either:
```suggestion
Mbed TLS provides an implementation of TLS 1.3. The TLS 1.3 support
```
or:
```suggestion
Mbed TLS provides an implementation of the TLS 1.3 protocol. The TLS 1.3 support
```
Would be better

> enablement is controlled by the MBEDTLS_SSL_PROTO_TLS1_3 configuration option.

Reviewer: Preexisting: The wording here is a bit tricky to understand. Perhaps something lik:
```suggestion
may be enabled using the MBEDTLS_SSL_PROTO_TLS1_3 configuration option.
```
